client.js

```diff
@@ -33,52 +35,103 @@ const cache = (args) => {
   })
 }
 
-const client = (args) => {
+const local = (command) => {
   return new Promise( (resolve, reject) => {
     let result = ''
     let resultError = ''
-    let child = spawn(`${blockchainCli} ${args.join(' ')}`, {
-      shell: '/bin/bash'
-    })
+    let child = spawn(command, { shell: '/bin/bash' })
 
     child.stderr.setEncoding('utf8')
     child.stdout.setEncoding('utf8')
 
+    child.stdout.on('data', (chunk) => {
+      result += chunk
+    })
+
+    child.stderr.on('data', (chunk) => {
+      resultError += chunk
+    })
+
     child.on('error', (err) => {
-      console.error(`errored with: ${err}`)
+      console.error(`Errored with: ${err}!`)
     })
 
     child.on('exit', (code, signal) => {
       if (code !== 0 || signal !== null) {
-        console.warn(`exited with code ${code} and signal: ${signal}`)
+        console.warn(`Exited with code ${code} and signal: ${signal}`)
       }
     })
 
     child.on('close', (reason) => {
       if (reason !== 0) {
-        console.warn(`closed for reason: ${reason}`)
+        console.warn(`Closed for reason: ${reason}`)
         reject(resultError)
       } else {
         resolve(result)
       }
     })
 
     child.on('message', (message) => {
-      console.log(`message recieved: ${message}`)
+      console.log(`Message recieved: ${message}`)
     })
+  })
+}
 
-    child.stdout.on('data', (data) => {
-      result += data
-    })
+// ToDo: will need to allow for POST (and perhaps other) types of requests too
+const remote = (request) => {
+  return new Promise( (resolve, reject) => {
+    https.get(request, (response) => {
+      let data = ''
 
-    child.stderr.on('data', (data) => {
-      resultError += data
+      response.setEncoding('utf8')
+
+      response.on('data', (chunk) => {
+        data += chunk
+      })
+
+      response.on('end', () => {
+        resolve(JSON.parse(data))
+      })
+    }).on('error', (err) => {
+      reject(err)
     })
   })
 }
 
+/**
+ *  This is the central component for spawning http requests or child processes which in turn query data from the user
+ *  defined blockchain (i.e. litecoin-cli or api.blockcypher.com).
+ *
+ *  [Spawns a promisified data stream using child process or https request based on "command" with optional "params"]
+ *  @param  {[array]} command :: the command/endpoint/url with required argument(s) -- required
+ *  @param  {[array]} params  :: optional parameters for the given command -- optional
+ *  @return {[promise]}       :: returns a promise; resolved when data stream ends, rejected on error
+ **/
+const client = (command, params) => {
+  if (typeof(params) !== 'undefined' && Array.isArray(params) === false) {
+    console.error(`Optional params must be an array! Instead got ${typeof(params)}`)
+    process.exit(1)
+  }
+
+  if (blockchainApi) {
+    let queryString = ''
+    if (Array.isArray(params)) {
+      queryString = `${command.join('/')}?${params.join('&')}`
```

Comment:
means need to pass parameters like so `['limit=100', 'txstart=30']` - not too bad albeit a bit clunky, alternatives: 
`{ limit: 500, txstart: 30 }` or even `[limit, 500, txstart, 30]`